contrib/lkt/extensions/mains/lkt_toolbox.adb

```diff
@@ -42,13 +43,27 @@ procedure Lkt_Toolbox is
 
    use Liblktlang;
 
+   package Invalid_Decl_Maps is new Ada.Containers.Hashed_Maps
+     (Key_Type        => Analysis.LK_Node,
+      Element_Type    => Boolean,
+      Hash            => Analysis.Hash,
+      Equivalent_Keys => "=");
+
    procedure Print_Semantic_Result
      (S : Analysis.Semantic_Result; Unit : Analysis.Analysis_Unit);
    --  Print a semantic result
 
    function Format_Node (Decl_Node : Decl'Class) return String;
    --  Format node for semantic result printing
 
+   function Populate_Invalid_Decl_Map
+      (Node : Analysis.LK_Node'Class) return Common.Visit_Status;
+   --  Populate invalid declarations map and check for nested invalid
+   --  declarations.
+
+   Invalid_Decl_Map : Invalid_Decl_Maps.Map;
+   --  Map of declarations annotated with @invalid
```

Comment:
This lacks a description of what the maps map to (i.e. the meaning of boolean there)